decredbuild.sh

```diff
@@ -1,65 +1,83 @@
 #!/bin/bash
 
+set -ex
+
 # Simple bash script to build basic Decred tools for all the platforms
 # we support with the golang cross-compiler.
 #
 # Copyright (c) 2016-2017 The Decred developers
 # Use of this source code is governed by the ISC
 # license.
 
-# If no tag specified, use date + version otherwise use tag.
-if [[ $1x = x ]]; then
-    DATE=`date +%Y%m%d`
-    VERSION="01"
-    TAG=$DATE-$VERSION
-else
-    TAG=$1
-    REL=(-ldflags "-X main.appBuild=release")
-    DCRWALLET_REL=(-ldflags "-X github.com/decred/dcrwallet/version.BuildMetadata=release")
+if [[ $# -ne 1 ]]; then
+	echo 'Please specify a version'
+	exit 1
+fi
+
+# GOPATH is currently required
+if [ -z "$GOPATH" ]; then
+	echo 'Please set GOPATH'
+	exit 1
 fi
 
+CWD=$PWD
+TAG=$1
 PACKAGE=decred
 MAINDIR=$PACKAGE-$TAG
+GO111MODULE=on
+REL=(-ldflags "-X github.com/decred/dcrd/internal/version.BuildMetadata=release -X github.com/decred/dcrd/internal/version.PreRelease=")
+DCRWALLET_REL=(-ldflags "-X github.com/decred/dcrwallet/version.BuildMetadata=release -X github.com/decred/dcrwallet/version.PreRelease=")
+
 mkdir -p $MAINDIR
-cd $MAINDIR
 
-SYS="windows-386 windows-amd64 openbsd-386 openbsd-amd64 linux-386 linux-amd64 linux-arm linux-arm64 darwin-amd64 dragonfly-amd64 freebsd-386 freebsd-amd64 freebsd-arm netbsd-386 netbsd-amd64 solaris-amd64"
+SYS="windows-386 windows-amd64 linux-386 linux-amd64 linux-arm linux-arm64 darwin-amd64 freebsd-amd64 openbsd-amd64"
 
-# Use the first element of $GOPATH in the case where GOPATH is a list
-# (something that is totally allowed).
-GPATH=$(echo $GOPATH | cut -f1 -d:)
+mkdir -p "$GOPATH/src/github.com/decred/"
```

Comment:
this seems dangerous, or at least unnecessary to build within GOPATH.  Can we do this elsewhere to always force a module build?